src/vm.rs

```diff
@@ -70,20 +71,20 @@ impl PevmTxExecutionResult {
     }
 }
 
-// TODO: Rewrite as [Result]
-pub(crate) enum VmExecutionResult {
+pub(crate) enum VmExecutionError {
     Retry,
     FallbackToSequential,
-    ReadError {
-        blocking_tx_idx: TxIdx,
-    },
+    ReadError { blocking_tx_idx: TxIdx },
     ExecutionError(ExecutionError),
-    Ok {
-        execution_result: PevmTxExecutionResult,
-        flags: FinishExecFlags,
-    },
 }
 
+pub(crate) struct VmExecutionOutcome {
+    pub execution_result: PevmTxExecutionResult,
+    pub flags: FinishExecFlags,
+}
+
+pub(crate) type VmExecutionResult = Result<VmExecutionOutcome, VmExecutionError>;
```

Comment:
Inline `Result<VmExecutionResult, VmExecutionError>` to the signature of `Vm::execute` is cleaner.